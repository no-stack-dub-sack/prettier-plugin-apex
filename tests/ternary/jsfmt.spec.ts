runSpec(
  __dirname,
  ["apex"],
  [
    { useTabs: false, tabWidth: 2 },
    { useTabs: false, tabWidth: 4 },
    { useTabs: true, tabWidth: 2 },
  ],
);
