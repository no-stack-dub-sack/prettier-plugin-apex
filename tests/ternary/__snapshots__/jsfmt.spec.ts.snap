// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Format apex: Ternary.cls: Ternary.cls 1`] = `
class Ternary {
  void ternaryWithMethodCall() {
    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod();
    longFunctionCallThatShouldBreakTheStatement(
      veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
    );
    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
      : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod()
      : anotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
      : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod();

      veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod(variableOne, variableTwo, variableThree, variableFour)
        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod(variableOne, variableTwo, variableThree, variableFour);

      List<String> a = (veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? new List<String>{ 'First One', 'Second One', 'Third One', 'Fourth One', 'Fifth One' }
        : new List<String>{ 'First One', 'Second One', 'Third One', 'Fourth One', 'Fifth One', 'Sixth One' });
  }

  void ternaryWithSoql() {
    List<Account> accounts = true
      ? [
          SELECT Id
          FROM Account
          WHERE
            Name = 'a name'
            OR Name = 'another name'
            OR Name = 'one more name'
        ]
      : [
          SELECT Id
          FROM Account
          WHERE
            Name = 'a name'
            OR Name = 'another name'
            OR Name = 'one more name'
        ];
  }
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class Ternary {
  void ternaryWithMethodCall() {
    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod();
    longFunctionCallThatShouldBreakTheStatement(
      veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
    );
    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
          ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
          : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
              ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
              : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod()
      : anotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
          ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
          : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
              ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
              : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod();

    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod(
          variableOne,
          variableTwo,
          variableThree,
          variableFour
        )
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod(
          variableOne,
          variableTwo,
          variableThree,
          variableFour
        );

    List<String> a = (veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? new List<String>{
          'First One',
          'Second One',
          'Third One',
          'Fourth One',
          'Fifth One'
        }
      : new List<String>{
          'First One',
          'Second One',
          'Third One',
          'Fourth One',
          'Fifth One',
          'Sixth One'
        });
  }

  void ternaryWithSoql() {
    List<Account> accounts = true
      ? [
          SELECT Id
          FROM Account
          WHERE
            Name = 'a name'
            OR Name = 'another name'
            OR Name = 'one more name'
        ]
      : [
          SELECT Id
          FROM Account
          WHERE
            Name = 'a name'
            OR Name = 'another name'
            OR Name = 'one more name'
        ];
  }
}

`;

exports[`Format apex: Ternary.cls: Ternary.cls 2`] = `
class Ternary {
  void ternaryWithMethodCall() {
    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod();
    longFunctionCallThatShouldBreakTheStatement(
      veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
    );
    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
      : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod()
      : anotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
      : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod();

      veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod(variableOne, variableTwo, variableThree, variableFour)
        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod(variableOne, variableTwo, variableThree, variableFour);

      List<String> a = (veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? new List<String>{ 'First One', 'Second One', 'Third One', 'Fourth One', 'Fifth One' }
        : new List<String>{ 'First One', 'Second One', 'Third One', 'Fourth One', 'Fifth One', 'Sixth One' });
  }

  void ternaryWithSoql() {
    List<Account> accounts = true
      ? [
          SELECT Id
          FROM Account
          WHERE
            Name = 'a name'
            OR Name = 'another name'
            OR Name = 'one more name'
        ]
      : [
          SELECT Id
          FROM Account
          WHERE
            Name = 'a name'
            OR Name = 'another name'
            OR Name = 'one more name'
        ];
  }
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class Ternary {
    void ternaryWithMethodCall() {
        veryLongVariableNameThatWillSurelyBreak == 'Yes'
            ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
            : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod();
        longFunctionCallThatShouldBreakTheStatement(
            veryLongVariableNameThatWillSurelyBreak == 'Yes'
                ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
                : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
        );
        veryLongVariableNameThatWillSurelyBreak == 'Yes'
            ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
                  ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
                  : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
                        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
                        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod()
            : anotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
                  ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
                  : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
                        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
                        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod();

        veryLongVariableNameThatWillSurelyBreak == 'Yes'
            ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod(
                  variableOne,
                  variableTwo,
                  variableThree,
                  variableFour
              )
            : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod(
                  variableOne,
                  variableTwo,
                  variableThree,
                  variableFour
              );

        List<String> a = (veryLongVariableNameThatWillSurelyBreak == 'Yes'
            ? new List<String>{
                  'First One',
                  'Second One',
                  'Third One',
                  'Fourth One',
                  'Fifth One'
              }
            : new List<String>{
                  'First One',
                  'Second One',
                  'Third One',
                  'Fourth One',
                  'Fifth One',
                  'Sixth One'
              });
    }

    void ternaryWithSoql() {
        List<Account> accounts = true
            ? [
                  SELECT Id
                  FROM Account
                  WHERE
                      Name = 'a name'
                      OR Name = 'another name'
                      OR Name = 'one more name'
              ]
            : [
                  SELECT Id
                  FROM Account
                  WHERE
                      Name = 'a name'
                      OR Name = 'another name'
                      OR Name = 'one more name'
              ];
    }
}

`;

exports[`Format apex: Ternary.cls: Ternary.cls 3`] = `
class Ternary {
  void ternaryWithMethodCall() {
    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod();
    longFunctionCallThatShouldBreakTheStatement(
      veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
    );
    veryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
      : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod()
      : anotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
      : yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
      ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
      : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod();

      veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod(variableOne, variableTwo, variableThree, variableFour)
        : VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod(variableOne, variableTwo, variableThree, variableFour);

      List<String> a = (veryLongVariableNameThatWillSurelyBreak == 'Yes'
        ? new List<String>{ 'First One', 'Second One', 'Third One', 'Fourth One', 'Fifth One' }
        : new List<String>{ 'First One', 'Second One', 'Third One', 'Fourth One', 'Fifth One', 'Sixth One' });
  }

  void ternaryWithSoql() {
    List<Account> accounts = true
      ? [
          SELECT Id
          FROM Account
          WHERE
            Name = 'a name'
            OR Name = 'another name'
            OR Name = 'one more name'
        ]
      : [
          SELECT Id
          FROM Account
          WHERE
            Name = 'a name'
            OR Name = 'another name'
            OR Name = 'one more name'
        ];
  }
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class Ternary {
	void ternaryWithMethodCall() {
		veryLongVariableNameThatWillSurelyBreak == 'Yes'
			? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
			: VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod();
		longFunctionCallThatShouldBreakTheStatement(
			veryLongVariableNameThatWillSurelyBreak == 'Yes'
				? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
				: VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
		);
		veryLongVariableNameThatWillSurelyBreak == 'Yes'
			? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod()
					? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
					: yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
							? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
							: VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod()
			: anotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
					? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod()
					: yetAnotherVeryLongVariableNameThatWillSurelyBreak == 'Yes'
							? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.yetAnotherMethod()
							: VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.finalAnotherMethod();

		veryLongVariableNameThatWillSurelyBreak == 'Yes'
			? VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.someMethod(
					variableOne,
					variableTwo,
					variableThree,
					variableFour
			  )
			: VeryLongClassNameThatWillForSureBeTooLongForThisMethodToNotBreak.anotherMethod(
					variableOne,
					variableTwo,
					variableThree,
					variableFour
			  );

		List<String> a = (veryLongVariableNameThatWillSurelyBreak == 'Yes'
			? new List<String>{
					'First One',
					'Second One',
					'Third One',
					'Fourth One',
					'Fifth One'
			  }
			: new List<String>{
					'First One',
					'Second One',
					'Third One',
					'Fourth One',
					'Fifth One',
					'Sixth One'
			  });
	}

	void ternaryWithSoql() {
		List<Account> accounts = true
			? [
					SELECT Id
					FROM Account
					WHERE
						Name = 'a name'
						OR Name = 'another name'
						OR Name = 'one more name'
			  ]
			: [
					SELECT Id
					FROM Account
					WHERE
						Name = 'a name'
						OR Name = 'another name'
						OR Name = 'one more name'
			  ];
	}
}

`;
